TYP:  ensure_platform_int probably needs a dtype annotation

`ensure_platform_int` is annotated as returning a `np.ndarray` but its output is used in places that require a `Sequence[int]`. I assume this can be fixed by adding a dtype to the `np.ndarray`.

pandas/core/groupby/ops.py:858: error: Argument 1 to "take" of "ExtensionArray" has incompatible type "ndarray[Any, Any]"; expected "Sequence[int]"  [arg-type]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,7 +216,7 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
-TakeIndexer = Union[Sequence[int], Sequence[np.integer] npt.NDArray[np.integer]]
+TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,7 +216,7 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
-TakeIndexer = Union[Sequence[ScalarIndexer], npt.NDArray[np.integer]]
+TakeIndexer = Union[Sequence[int], Sequence[np.integer] npt.NDArray[np.integer]]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -222,7 +222,7 @@ PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]
 if TYPE_CHECKING:
     TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
 else:
-    TakeIndexer = Union[Sequence[int], Sequence[np.integer], np.ndarray]
+    TakeIndexer = Any
 
 # Windowing rank methods
 WindowingRankType = Literal["average", "min", "max"]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,10 +216,13 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
-TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]
+if npt is not None:
+    TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
+else:
+    TakeIndexer = Union[Sequence[int], npt.NDArray[np.integer]]
 
 # Windowing rank methods
 WindowingRankType = Literal["average", "min", "max"]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -222,7 +222,7 @@ PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]
 if npt is not None:
     TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
 else:
-    TakeIndexer = Union[Sequence[int], npt.NDArray[np.integer]]
+    TakeIndexer = Union[Sequence[int], Sequence[np.integer], np.ndarray]
 
 # Windowing rank methods
 WindowingRankType = Literal["average", "min", "max"]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,7 +216,7 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
-GeneralSequenceIndexer = Union[Sequence[int], SequenceIndexer]
+TakeIndexer = Union[Sequence[int], np.ndarray]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -219,7 +219,7 @@ SequenceIndexer = Union[slice, List[int], np.ndarray]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]
-if npt is not None:
+if TYPE_CHECKING:
     TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
 else:
     TakeIndexer = Union[Sequence[int], Sequence[np.integer], np.ndarray]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,7 +216,7 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
-TakeIndexer = Union[Sequence[int], np.ndarray]
+TakeIndexer = Union[Sequence[ScalarIndexer], npt.NDArray[np.integer]]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]

--- a/pandas/_typing.py
+++ b/pandas/_typing.py
@@ -216,6 +216,7 @@ Manager2D = Union["ArrayManager", "BlockManager"]
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
+GeneralSequenceIndexer = Union[Sequence[int], SequenceIndexer]
 PositionalIndexer = Union[ScalarIndexer, SequenceIndexer]
 PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
 PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -27,7 +27,7 @@ from pandas._typing import (
     AnyArrayLike,
     ArrayLike,
     DtypeObj,
-    PositionalIndexer,
+    NonScalarPositionalIndexer,
     Scalar,
     npt,
 )
@@ -1433,7 +1433,7 @@ class SelectNFrame(SelectN):
 
 def take(
     arr,
-    indices: PositionalIndexer,
+    indices: NonScalarPositionalIndexer,
     axis: int = 0,
     allow_fill: bool = False,
     fill_value=None,

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -27,8 +27,8 @@ from pandas._typing import (
     AnyArrayLike,
     ArrayLike,
     DtypeObj,
-    NonScalarPositionalIndexer,
     Scalar,
+    SequenceIndexer,
     npt,
 )
 from pandas.util._decorators import doc
@@ -1433,7 +1433,7 @@ class SelectNFrame(SelectN):
 
 def take(
     arr,
-    indices: NonScalarPositionalIndexer,
+    indices: SequenceIndexer,
     axis: int = 0,
     allow_fill: bool = False,
     fill_value=None,

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -27,6 +27,7 @@ from pandas._typing import (
     AnyArrayLike,
     ArrayLike,
     DtypeObj,
+    PositionalIndexer,
     Scalar,
     npt,
 )
@@ -1431,7 +1432,11 @@ class SelectNFrame(SelectN):
 
 
 def take(
-    arr, indices: np.ndarray, axis: int = 0, allow_fill: bool = False, fill_value=None
+    arr,
+    indices: PositionalIndexer,
+    axis: int = 0,
+    allow_fill: bool = False,
+    fill_value=None,
 ):
     """
     Take elements from an array.

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -27,6 +27,7 @@ from pandas._typing import (
     AnyArrayLike,
     ArrayLike,
     DtypeObj,
+    GeneralSequenceIndexer,
     Scalar,
     SequenceIndexer,
     npt,
@@ -1433,7 +1434,7 @@ class SelectNFrame(SelectN):
 
 def take(
     arr,
-    indices: SequenceIndexer,
+    indices: GeneralSequenceIndexer,
     axis: int = 0,
     allow_fill: bool = False,
     fill_value=None,

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -1446,7 +1446,7 @@ def take(
     arr : array-like or scalar value
         Non array-likes (sequences/scalars without a dtype) are coerced
         to an ndarray.
-    indices : sequence of integers
+    indices : sequence of int or one-dimensional np.ndarray of int
         Indices to be taken.
     axis : int, default 0
         The axis over which to select values.

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -29,7 +29,6 @@ from pandas._typing import (
     DtypeObj,
     GeneralSequenceIndexer,
     Scalar,
-    SequenceIndexer,
     npt,
 )
 from pandas.util._decorators import doc

--- a/pandas/core/algorithms.py
+++ b/pandas/core/algorithms.py
@@ -27,8 +27,8 @@ from pandas._typing import (
     AnyArrayLike,
     ArrayLike,
     DtypeObj,
-    GeneralSequenceIndexer,
     Scalar,
+    TakeIndexer,
     npt,
 )
 from pandas.util._decorators import doc
@@ -1433,7 +1433,7 @@ class SelectNFrame(SelectN):
 
 def take(
     arr,
-    indices: GeneralSequenceIndexer,
+    indices: TakeIndexer,
     axis: int = 0,
     allow_fill: bool = False,
     fill_value=None,

--- a/pandas/core/array_algos/take.py
+++ b/pandas/core/array_algos/take.py
@@ -178,14 +178,7 @@ def take_1d(
     """
     if not isinstance(arr, np.ndarray):
         # ExtensionArray -> dispatch to their method
-
-        # error: Argument 1 to "take" of "ExtensionArray" has incompatible type
-        # "ndarray"; expected "Sequence[int]"
-        return arr.take(
-            indexer,  # type: ignore[arg-type]
-            fill_value=fill_value,
-            allow_fill=allow_fill,
-        )
+        return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
 
     if not allow_fill:
         return arr.take(indexer)

--- a/pandas/core/arrays/_mixins.py
+++ b/pandas/core/arrays/_mixins.py
@@ -17,6 +17,7 @@ from pandas._libs import lib
 from pandas._libs.arrays import NDArrayBacked
 from pandas._typing import (
     F,
+    GeneralSequenceIndexer,
     PositionalIndexer2D,
     PositionalIndexerTuple,
     ScalarIndexer,
@@ -101,7 +102,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
     def take(
         self: NDArrayBackedExtensionArrayT,
-        indices: SequenceIndexer,
+        indices: GeneralSequenceIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/_mixins.py
+++ b/pandas/core/arrays/_mixins.py
@@ -17,6 +17,7 @@ from pandas._libs import lib
 from pandas._libs.arrays import NDArrayBacked
 from pandas._typing import (
     F,
+    PositionalIndexer,
     PositionalIndexer2D,
     PositionalIndexerTuple,
     ScalarIndexer,
@@ -101,7 +102,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
     def take(
         self: NDArrayBackedExtensionArrayT,
-        indices: Sequence[int],
+        indices: PositionalIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,
@@ -112,9 +113,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
         new_data = take(
             self._ndarray,
-            # error: Argument 2 to "take" has incompatible type "Sequence[int]";
-            # expected "ndarray"
-            indices,  # type: ignore[arg-type]
+            indices,
             allow_fill=allow_fill,
             fill_value=fill_value,
             axis=axis,

--- a/pandas/core/arrays/_mixins.py
+++ b/pandas/core/arrays/_mixins.py
@@ -17,7 +17,7 @@ from pandas._libs import lib
 from pandas._libs.arrays import NDArrayBacked
 from pandas._typing import (
     F,
-    PositionalIndexer,
+    NonScalarPositionalIndexer,
     PositionalIndexer2D,
     PositionalIndexerTuple,
     ScalarIndexer,
@@ -102,7 +102,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
     def take(
         self: NDArrayBackedExtensionArrayT,
-        indices: PositionalIndexer,
+        indices: NonScalarPositionalIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/_mixins.py
+++ b/pandas/core/arrays/_mixins.py
@@ -17,12 +17,12 @@ from pandas._libs import lib
 from pandas._libs.arrays import NDArrayBacked
 from pandas._typing import (
     F,
-    GeneralSequenceIndexer,
     PositionalIndexer2D,
     PositionalIndexerTuple,
     ScalarIndexer,
     SequenceIndexer,
     Shape,
+    TakeIndexer,
     npt,
     type_t,
 )
@@ -102,7 +102,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
     def take(
         self: NDArrayBackedExtensionArrayT,
-        indices: GeneralSequenceIndexer,
+        indices: TakeIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/_mixins.py
+++ b/pandas/core/arrays/_mixins.py
@@ -17,7 +17,6 @@ from pandas._libs import lib
 from pandas._libs.arrays import NDArrayBacked
 from pandas._typing import (
     F,
-    NonScalarPositionalIndexer,
     PositionalIndexer2D,
     PositionalIndexerTuple,
     ScalarIndexer,
@@ -102,7 +101,7 @@ class NDArrayBackedExtensionArray(NDArrayBacked, ExtensionArray):
 
     def take(
         self: NDArrayBackedExtensionArrayT,
-        indices: NonScalarPositionalIndexer,
+        indices: SequenceIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -29,6 +29,7 @@ from pandas._typing import (
     AstypeArg,
     Dtype,
     FillnaOptions,
+    GeneralSequenceIndexer,
     PositionalIndexer,
     ScalarIndexer,
     SequenceIndexer,
@@ -1076,7 +1077,7 @@ class ExtensionArray:
 
     def take(
         self: ExtensionArrayT,
-        indices: SequenceIndexer,
+        indices: GeneralSequenceIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -29,6 +29,7 @@ from pandas._typing import (
     AstypeArg,
     Dtype,
     FillnaOptions,
+    NonScalarPositionalIndexer,
     PositionalIndexer,
     ScalarIndexer,
     SequenceIndexer,
@@ -1076,7 +1077,7 @@ class ExtensionArray:
 
     def take(
         self: ExtensionArrayT,
-        indices: PositionalIndexer,
+        indices: NonScalarPositionalIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -1087,7 +1087,7 @@ class ExtensionArray:
 
         Parameters
         ----------
-        indices : sequence of int
+        indices : sequence of int or one-dimensional np.ndarray of int
             Indices to be taken.
         allow_fill : bool, default False
             How to handle negative values in `indices`.

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -29,7 +29,6 @@ from pandas._typing import (
     AstypeArg,
     Dtype,
     FillnaOptions,
-    NonScalarPositionalIndexer,
     PositionalIndexer,
     ScalarIndexer,
     SequenceIndexer,
@@ -1077,7 +1076,7 @@ class ExtensionArray:
 
     def take(
         self: ExtensionArrayT,
-        indices: NonScalarPositionalIndexer,
+        indices: SequenceIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -29,11 +29,11 @@ from pandas._typing import (
     AstypeArg,
     Dtype,
     FillnaOptions,
-    GeneralSequenceIndexer,
     PositionalIndexer,
     ScalarIndexer,
     SequenceIndexer,
     Shape,
+    TakeIndexer,
     npt,
 )
 from pandas.compat import set_function_name
@@ -1077,7 +1077,7 @@ class ExtensionArray:
 
     def take(
         self: ExtensionArrayT,
-        indices: GeneralSequenceIndexer,
+        indices: TakeIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/base.py
+++ b/pandas/core/arrays/base.py
@@ -1076,7 +1076,7 @@ class ExtensionArray:
 
     def take(
         self: ExtensionArrayT,
-        indices: Sequence[int],
+        indices: PositionalIndexer,
         *,
         allow_fill: bool = False,
         fill_value: Any = None,

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -521,7 +521,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
 
         Parameters
         ----------
-        indices : sequence of int
+        indices : sequence of int or one-dimensional np.ndarray of int
             Indices to be taken.
         allow_fill : bool, default False
             How to handle negative values in `indices`.

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -307,9 +307,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
             if not len(item):
                 return type(self)(pa.chunked_array([], type=pa.string()))
             elif is_integer_dtype(item.dtype):
-                # error: Argument 1 to "take" of "ArrowStringArray" has incompatible
-                # type "ndarray"; expected "Sequence[int]"
-                return self.take(item)  # type: ignore[arg-type]
+                return self.take(item)
             elif is_bool_dtype(item.dtype):
                 return type(self)(self._data.filter(item))
             else:
@@ -513,7 +511,10 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
                 self[k] = v
 
     def take(
-        self, indices: Sequence[int], allow_fill: bool = False, fill_value: Any = None
+        self,
+        indices: PositionalIndexer,
+        allow_fill: bool = False,
+        fill_value: Any = None,
     ):
         """
         Take elements from an array.

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -18,6 +18,7 @@ from pandas._libs import (
 )
 from pandas._typing import (
     Dtype,
+    GeneralSequenceIndexer,
     NpDtype,
     PositionalIndexer,
     Scalar,
@@ -511,7 +512,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
 
     def take(
         self,
-        indices: SequenceIndexer,
+        indices: GeneralSequenceIndexer,
         allow_fill: bool = False,
         fill_value: Any = None,
     ):

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -5,7 +5,6 @@ import re
 from typing import (
     TYPE_CHECKING,
     Any,
-    Sequence,
     Union,
     cast,
     overload,

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -18,12 +18,12 @@ from pandas._libs import (
 )
 from pandas._typing import (
     Dtype,
-    GeneralSequenceIndexer,
     NpDtype,
     PositionalIndexer,
     Scalar,
     ScalarIndexer,
     SequenceIndexer,
+    TakeIndexer,
     npt,
 )
 from pandas.compat import (
@@ -512,7 +512,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
 
     def take(
         self,
-        indices: GeneralSequenceIndexer,
+        indices: TakeIndexer,
         allow_fill: bool = False,
         fill_value: Any = None,
     ):

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -18,6 +18,7 @@ from pandas._libs import (
 )
 from pandas._typing import (
     Dtype,
+    NonScalarPositionalIndexer,
     NpDtype,
     PositionalIndexer,
     Scalar,
@@ -511,7 +512,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
 
     def take(
         self,
-        indices: PositionalIndexer,
+        indices: NonScalarPositionalIndexer,
         allow_fill: bool = False,
         fill_value: Any = None,
     ):

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -18,7 +18,6 @@ from pandas._libs import (
 )
 from pandas._typing import (
     Dtype,
-    NonScalarPositionalIndexer,
     NpDtype,
     PositionalIndexer,
     Scalar,
@@ -512,7 +511,7 @@ class ArrowStringArray(OpsMixin, BaseStringArray, ObjectStringArrayMixin):
 
     def take(
         self,
-        indices: NonScalarPositionalIndexer,
+        indices: SequenceIndexer,
         allow_fill: bool = False,
         fill_value: Any = None,
     ):

--- a/pandas/core/indexes/base.py
+++ b/pandas/core/indexes/base.py
@@ -4318,12 +4318,8 @@ class Index(IndexOpsMixin, PandasObject):
         )
         mask = left_idx == -1
 
-        # error: Argument 1 to "take" of "ExtensionArray" has incompatible
-        # type "ndarray[Any, dtype[signedinteger[Any]]]"; expected "Sequence[int]"
-        join_array = self._values.take(left_idx)  # type: ignore[arg-type]
-        # error: Argument 1 to "take" of "ExtensionArray" has incompatible type
-        # "ndarray[Any, dtype[signedinteger[Any]]]"; expected "Sequence[int]"
-        right = other._values.take(right_idx)  # type: ignore[arg-type]
+        join_array = self._values.take(left_idx)
+        right = other._values.take(right_idx)
 
         if isinstance(join_array, np.ndarray):
             np.putmask(join_array, mask, right)

